compiler/src/iree/compiler/Codegen/Dialect/VectorExt/IR/test/roundtrip.mlir

```diff
@@ -88,6 +88,37 @@ func.func @specify_nested(%lhs: memref<32x32xf16>) -> vector<32x32xf16> {
 
 // -----
 
+#nested_0 = #iree_vector_ext.nested_layout<
+  subgroup_tile = [],
+  batch_tile = [],
+  outer_tile = [],
+  thread_tile = [],
+  element_tile = [],
+
+  subgroup_strides = [],
+  thread_strides   = []
+>
+
+func.func @specify_nested_0d(%lhs: vector<f16>) -> vector<f16> {
+  %result = iree_vector_ext.to_layout %lhs to layout(#nested_0) : vector<f16>
+  func.return %result : vector<f16>
+}
+
+// CHECK: #[[$LAYOUT0:.+]] = #iree_vector_ext.nested_layout<
```

Comment:
Doesnt the `OptionalArrayRefParamater` imply that any of these could be dropped from the input IR? I dont think the lit test maps to the change here.

Reply:
No, the assembly format requires for them to be there in the IR. The verifier also checks that they all have the same rank. "OptionalArrayRefParameter" allows empty array refs, everything else is same (it even has a backing allocator with SmallVector<int64_t>.